app/services/person_service.rb

```diff
@@ -180,7 +180,7 @@ def soundex_person_search(given_name, middle_name, family_name, gender)
       # soundex, vice-versa is also true, thus we capture both then combine them.
       filters = { given_name:, middle_name:, family_name: }
       raw_matches = NameSearchService.search_full_person_name(filters, use_soundex: false)
-      soundex_matches = NameSearchService.search_full_person_name(filters, use_soundex: true)
+      soundex_matches = NameSearchService.search_full_person_name(filters, use_soundex: false)
```

Comment:
I am thinking let us fix the soundex this will cause us to miss some clients

Reply:
@fchiyenda  I have made the following changes:

-First, i call search_full_raw_person_name with the provided filters to perform a raw match search.
 If the raw match search returns results, it returns the found results.
 If no results are found, it falls back to calling search_full_soundex_person_name with the same filters to perform a soundex search but the soundex function i have also made few changes 
 - to generate soundex codes for both the full value and its partial prefix
 - search using both the full soundex and partial soundex codes
 This way it ensures the results are returned to the user faster

Please review the updates at your convenience.